src/burn/engine/plan.cpp

```diff
@@ -2742,6 +2764,84 @@ static HRESULT FinalizeSlipstreamPatchActions(
     return hr;
 }
 
+static HRESULT PlanDependencyActions(
```

Comment:
This change will be necessary for the other change to work fully. @rseanhall can you port this change to v3.x to get the `BOOTSTRAPPER_ACTION_CACHE` using it?